lib/addons/cloudwatch-adot-addon/index.ts

```diff
@@ -0,0 +1,80 @@
+import { KubernetesManifest } from "aws-cdk-lib/aws-eks";
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { dependable, loadYaml, readYamlDocument } from "../../utils";
+import { AdotCollectorAddOn } from "../adot";
+import { Construct } from 'constructs';
+
+/**
+ * Configuration options for add-on.
+ */
+ export interface CloudWatchAdotAddOnProps {
+    /**
+     * Modes supported : `deployment`, `daemonset`, `statefulSet`, and `sidecar`
+     * @default deployment
+     */
+    deploymentMode?: cloudWatchDeploymentMode;
+    /**
+     * Namespace to deploy the ADOT Collector for CloudWatch.
+     * @default default
+     */
+    namepace?: string;
+}
+
+export const enum cloudWatchDeploymentMode {
+    DEPLOYMENT = 'deployment',
+    DAEMONSET = 'daemonset',
+    STATEFULSET = 'statefulset',
+    SIDECAR = 'sidecar'
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps = {
+    deploymentMode: cloudWatchDeploymentMode.DEPLOYMENT,
+    namespace: 'default'
+};
+
+/**
+ * Implementation of CloudWatch ADOT add-on for EKS Blueprints. Installs ADOT Collector.
+ */
+export class CloudWatchAdotAddOn implements ClusterAddOn {
+
+    readonly cloudWatchAddOnProps: CloudWatchAdotAddOnProps;
+
+    constructor(props?: CloudWatchAdotAddOnProps) {
+        this.cloudWatchAddOnProps = { ...defaultProps, ...props };
+    }
+
+    @dependable(AdotCollectorAddOn.name)
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let finalDeploymentMode: string;
+        let finalNamespace: string;
+        let doc: string;
+        
+        finalDeploymentMode = defaultProps.deploymentMode;
+        if (this.cloudWatchAddOnProps.deploymentMode) {
+            finalDeploymentMode = this.cloudWatchAddOnProps.deploymentMode;
+        }
+
+        finalNamespace = defaultProps.namespace;
+        if (this.cloudWatchAddOnProps.namepace) {
+            finalNamespace = this.cloudWatchAddOnProps.namepace;
+        }
+
+        // Applying manifest for configuring ADOT Collector for CloudWatch.
+        doc = readYamlDocument(__dirname + '/collector-config-cloudwatch.ytpl');
```

Comment:
Can we use https://github.com/aws-observability/aws-otel-helm-charts/tree/main/charts/adot-exporter-for-eks-on-ec2 helm chart for this? 
I am hesitant to apply arbitrary yaml files due to lack of continuity of such support.

Reply:
This Addon approach is not same as `adot-exporter-for-eks-on-ec2` which is used in Container Insights Addon. CW Addon uses `adot` as an operator which manages the lifecycle of the `adot` collector which is a better approach vs managing the `adot` collector via deployments and statefulsets. We should give flexibility to customers to use both the approaches as needed and other one can be deprecated later. I understand the hesitancy but yaml approach is what we as an approach in ADOT product documentation. I will check with oTel team on when they are doing to come up with helm chart.

Reply:
I will setup a meeting to discuss on this.